### PR TITLE
Store: acceptance test for proxy store

### PR DESCRIPTION
**Pull request by MichaHoffmann**
* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

* Add basic acceptance tests for proxy store
* Fix bug where invalid requests got ignored because of partial response strategy

## Verification

* new acceptance tests for proxy store

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,8 @@ We use *breaking :warning:* to mark changes that are not backward compatible (re
 
 ### Fixed
 
+- [#7084](https://github.com/thanos-io/thanos/pull/7084) Store: fix parameter validation for proxy store
```
Comment (GiedriusS) [inline CHANGELOG.md:15]: This is good but I think it's impossible to trigger this due to the check here https://github.com/thanos-io/thanos/blob/main/pkg/api/query/v1.go#L1034 so IMHO not worth mentioning it here.

**Comment by MichaHoffmann** [comment]
> > Fix bug where invalid requests got ignored because of partial response strategy
> 
> Where is this fix? 😄

Its this one: 
```
	if r.Label == "" {
		return nil, status.Error(codes.InvalidArgument, "label name parameter cannot be empty")
	}
```

File: CHANGELOG.md (hunk 1 above)
Comment (MichaHoffmann) [reply CHANGELOG.md:15]: Ah yeah! I just noticed it in acceptance tests but not a user facing issue, yes!

**Review by douglascamata — APPROVED** [review]
:shipit:

**Pull request merged**